### PR TITLE
Plot margins

### PR DESCRIPTION
**Pull request by thomased**

### CODE REVIEW

**Comment by thomased** [comment]
I've left `jndplot()` alone for now since I _think_ it needs it. But otherwise the margin-fiddling is all removed. Some of the default plots look a touch sub-optimal as a result, but like you say that's up to users, as is the case with all plots.

File: R/tetraplot.R [hunk 1]
```diff
@@ -186,7 +186,7 @@ tetraplot <- function(tcsdata, theta = 45, phi = 10, perspective = FALSE,
   }
 
 
-  par(mar = margin, pty = "s", new = TRUE)
+  par(pty = "s", new = TRUE)
```
Comment (Bisaloo) [inline R/tetraplot.R:189]: The `par(pty = "s")` is necessary here and cannot be replaced by `asp = 1`?
If so, it'd be helpful to add a comment to explain why.

File: R/plot.colspace.R [hunk 2]
```diff
@@ -74,8 +74,8 @@
 
 plot.colspace <- function(x, ...) {
 
-  oldpar <- par(no.readonly = TRUE)
-  on.exit(par(oldpar))
+  # oldpar <- par(no.readonly = TRUE)
+  # on.exit(par(oldpar))
```
Comment (Bisaloo) [inline R/plot.colspace.R:79]:
```suggestion
```

**Review by Bisaloo — COMMENTED** [review]
I didn't test it just yet but just a couple of comments while browsing the code.

Additionally, I still see `par(margins` in `explorespec()` and `projplot()`. Is it necessary there as well?

File: R/tetraplot.R (hunk 1 above)
Comment (thomased) [reply R/tetraplot.R]: Nope, just didn't search thoroughly enough...

`asp = 1` produces better results anyway.

**Comment by thomased** [comment]
> I didn't test it just yet but just a couple of comments while browsing the code.
> 
> Additionally, I still see `par(margins` in `explorespec()` and `projplot()`. Is it necessary there as well?

`projplot()` no, cheers — fixed that. But I'm inclined to keep `explorespec()` as-is. Because it's multi-panel in nature I think it's simplest if we just set nice margins. And it's not a publication-grade plot, it's really just for diagnostic purposes.

File: R/cieplot.R [hunk 3]
```diff
@@ -53,11 +52,12 @@
...
                     out.lty = 1, theta = 45, phi = 10, r = 1e6, zoom = 1, box = FALSE,
-                    margin = c(0, 0, 0, 0), ciebg = TRUE, ...) {
-
-  oldpar <- par(no.readonly = TRUE)
-  on.exit(par(oldpar))
+                    ciebg = TRUE, margin = NULL, ...) {
 
+  if (!missing("margin"))
+    message("The 'margin' argument is deprecated, and will be ignored. See ?par() for guidance on 
+            setting margins in the standard manner.")
```
Comment (Bisaloo) [inline R/cieplot.R:59]: The alignment might not be what you expect here. Can you double check if that's what you want? Below is a proposal of what I think you meant:

```suggestion
    message(
      "The 'margin' argument is deprecated, and will be ignored. ", 
      "See ?par() for guidance on setting margins in the standard manner."
    )
```

On a related note, shouldn't this be a warning?

File: NEWS.md [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 * Fixed a recent bug whereby the D65 illuminant was expressed in units of radiant flux, by default. It has instead been converted to photon flux (umol.s-1.m-2). 
 * Added an informative error message when raw quantum catch estimates are < 1, and an attempt is made to calculate colour distances in the receptor-noise model using noise = 'quantum'. Taking the root of negative values following log-transformation, as required when noise = 'quantum', would previously produce an uninformative error, which has been rectified. 
+* The `margins` argument has been deprecated for cie, coc, hexagon, segment, projplot, di-, tri-, and tetra-chromatic plots. It was causing problems when later adding points to plots, and margin-handling is best left to the user anyway. The default plots may therefore look slightly different, but the margins can be adjusted to taste using the standard par(mar = c()) pathway.
```
Comment (Bisaloo) [inline NEWS.md:7]:
```suggestion
* The `margins` argument has been deprecated for cie, coc, hexagon, segment, projplot, di-, tri-, and tetra-chromatic plots. It was causing problems when later adding points to plots, and margin-handling is best left to the user anyway. The default plots may therefore look slightly different, but the margins can be adjusted to taste using the standard `par(mar = c())` pathway.
```

(for auto-linking on the pkgdown website)

File: R/cieplot.R [hunk 5]
```diff
@@ -19,10 +19,9 @@
 #'   from the plotting area when `space = "cielab"`.
 #' @param box logical. Should the plot area box and axes be plotted? (defaults
 #'   to `FALSE`)
-#' @param margin vector of four numbers specifying drawing margins for CIELAB
-#'   plot (defaults to c(0,0,0,0))
 #' @param ciebg should the colour background be plotted for CIEXYZ plot?
 #'   (defaults to `TRUE`)
+#' @param margin Deprecated. Please use the standard par() method for custom margins.
```
Comment (Bisaloo) [inline R/cieplot.R:24]:
```suggestion
#' @param margin Deprecated. Please use the standard [par()] method for custom margins. 
```

for formatting and auto-linking

File: R/tetraplot.R [hunk 6]
```diff
@@ -25,10 +25,9 @@
 #'   `TRUE`)
 #' @param vert.cex size of the points at the vertices (defaults to 1).
 #' @param out.lwd,out.lcol graphical parameters for the tetrahedral outline.
-#' @param margin vector of four numbers specifying drawing margins (defaults to
-#'   c(0, 0, 0, 0)).
 #' @param type accepts a vector of length 1 or 2 with 'p' for points and/or 'l'
 #'   for lines from the point to the base of the tetrahedron.
+#' @param margin Deprecated. Please use the standard par() method for custom margins.
```
Comment (Bisaloo) [inline R/tetraplot.R:30]:
```suggestion
#' @param margin Deprecated. Please use the standard [par()] method for custom margins. 
```

File: R/triplot.R [hunk 7]
```diff
@@ -14,12 +14,12 @@
 #' @param achrocol color of the point at the origin `achro = TRUE` (defaults to
 #'   `'grey'`).
 #' @param out.lwd,out.lcol,out.lty graphical parameters for the plot outline.
-#' @param margins margins for the plot.
 #' @param square logical. Should the aspect ratio of the plot be held to 1:1?
 #'   (defaults to `TRUE`).
 #' @param gamut logical. Should the polygon showing the possible colours given
 #'   visual system and illuminant used in the analysis (defaults to `FALSE`).
 #'   This option currently only works when `qcatch = Qi`.
+#' @param margins Deprecated. Please use the standard par() method for custom margins.
```
Comment (Bisaloo) [inline R/triplot.R:22]:
```suggestion
#' @param margins Deprecated. Please use the standard [par()] method for custom margins. 
```

**Review by Bisaloo — COMMENTED** [review]
Looks very good to me. Thank you for this!

Couple of new comments:
- `legendtetra()` seems broken but I don't know if it's related to these changes
- we will have to re-roxygenise after merging the last round of changes
- for the vignettes, it might be nice to crop the plots to remove the extra blank space in some cases. Instructions: https://bookdown.org/yihui/rmarkdown-cookbook/crop-plot.html. We should wait until we see the output in pkgdown IMO as it will be easier to spots possible problems but as you wish.

**Comment by thomased** [comment]
Awesome — appreciate it. Yeah I had the same thought about the vignettes but agree it'll be easiest once merged in.

**Comment by thomased** [comment]
And no, `legendtetra()` appears broken on `master` too. Odd. There's an issue relating to the function more generally (#54) already, we can continue thinking about it there.

**Pull request merged**